pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClient.java

```diff
@@ -94,16 +103,13 @@ public void afterBulk(long executionId, List<BulkProcessor.BulkOperationRequest>
         };
         this.backoffRetry = new RandomExponentialRetry(elasticSearchConfig.getMaxRetryTimeInSec());
         this.client = retry(() -> RestClientFactory.createClient(config, bulkListener), -1, "client creation");
+        state.set(State.Open);
     }
 
     void failed(Exception e) {
-        if (irrecoverableError.compareAndSet(null, e)) {
-            log.error("Irrecoverable error:", e);
-        }
-    }
-
-    boolean isFailed() {
-        return irrecoverableError.get() != null;
+        state.set(State.Failed);
```

Comment:
we should call fatal and close only once

```
if (state.compareAndSet(State.Open, State.Failed)) {
  sinkContext.fatal(e);
  close();
}
```

Reply:
Thanks. Fixed it.